qualtran/bloqs/rotations/hamming_weight_phasing_test.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import cirq
+import numpy as np
+import pytest
+
+from qualtran.bloqs.rotations.hamming_weight_phasing import HammingWeightPhasing
+from qualtran.cirq_interop.testing import GateHelper
+from qualtran.testing import assert_valid_bloq_decomposition
+
+
```

Comment:
One more thing, since the Toffoli complexity is known (and alpha is also known) is it easy to check that the  is correct either through t_complexity or the call_graph?

Reply:
Changed the `gate.t_complexity().t <= (n - 1)` assertion to ``gate.t_complexity().t == (n - n.bit_count())`` to make it more robust.